crates/sui-framework/sources/governance/sui_system.move

```diff
@@ -148,9 +145,8 @@ module sui::sui_system {
             // Use a hardcoded ID.
             id: object::sui_system_state(),
             version: protocol_version,
-            system_state,
         };
-        // dynamic_object_field::add(&mut self.id, self.version, system_state);
+        dynamic_field::add(&mut self.id, protocol_version, system_state);
```

Comment:
i'm confused by this - if we advance the protocol version, how do we find this field afterward? I don't see any code yet that migrates the inner state from one version to the next.

Reply:
This is genesis, and I am making the initial version of system state the same as protocol version. 
We could also make them constant, if that's less confusing

Reply:
And upgrade logic hasn't been added yet